stl/inc/bit

```diff
@@ -233,6 +233,34 @@ _NODISCARD int _Checked_arm_arm64_countl_zero(const _Ty _Val) noexcept {
 }
 #endif // defined(_M_ARM) || defined(_M_ARM64)
 
+#if _HAS_CXX23
+_NODISCARD constexpr unsigned short _Byteswap_ushort(const unsigned short _Val) noexcept {
+    if (_STD is_constant_evaluated()) {
+        return static_cast<unsigned short>((_Val << 8) | (_Val >> 8));
+    } else {
+        return _byteswap_ushort(_Val);

```

Comment:
No change requested - I was initially wondering how this was compiling. I observed that these intrinsics are always available for non-`/clr:pure` code (they're `__MACHINE`, which is good - it means "all architectures"), but they're declared in `intrin.h` instead of `intrin0.h`:

```
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include>grep -rP byteswap *
intrin.h:__MACHINE(_NODISCARD _Check_return_ unsigned __int64 __cdecl _byteswap_uint64(_In_ unsigned __int64))
intrin.h:__MACHINE(_NODISCARD _Check_return_ unsigned long __cdecl _byteswap_ulong(_In_ unsigned long))
intrin.h:__MACHINE(_NODISCARD _Check_return_ unsigned short __cdecl _byteswap_ushort(_In_ unsigned short))
```

After compiling the test with `/P` and `/showIncludes`, I found out why:

`<bit>` includes `<limits>` includes `<xstddef>` includes `<cstdlib>` includes `<stdlib.h>`, which declares these intrinsics, and [is documented as doing so](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/byteswap-uint64-byteswap-ulong-byteswap-ushort?view=msvc-160).

Reply:
Mea culpa - I should have documented this and saved you the trouble of diving down this particular hole.